wallet/wallet.go

```diff
@@ -317,6 +317,22 @@ func (w *Wallet) activeData(dbtx walletdb.ReadTx) ([]btcutil.Address, []wtxmgr.C
 	return addrs, unspent, err
 }
 
+func (w *Wallet) onClientConnected(birthdayStamp *waddrmgr.BlockStamp) {
+	w.wg.Add(1)
```

Comment:
this should be called the line before the goroutine

Reply:
done!

Reply:
@halseth another issue with the syncWithChain function related to the shutdown is that it runs in a loop and doesn't check for ShuttingDown() during its run. The problem even get worst when the shutdown request comes at time where the code is at this point: https://github.com/btcsuite/btcwallet/blob/master/wallet/wallet.go#L460
the go routine keeps running forever stuck inside this block preventing from wallet shutdown.
I can add such kind of check inside this block and on every iteration of the outer loop. Do you think it is a good solution or can suggest a better one?

Reply:
Yeah, checking the quit signal on every iteration sounds like  a good idea :)